app/models/consent_form.rb

```diff
@@ -342,26 +340,41 @@ def ask_for_contact_method?
     Flipper.enabled?(:parent_contact_method) && parent_phone.present?
   end
 
-  def seed_health_questions_if_consent_given
-    return unless consent_given? && health_answers.empty?
-    vaccine = campaign.vaccines.first
-    self.health_answers = vaccine.health_questions.to_health_answers
-  end
-
-  def remove_health_questions_if_consent_refused
-    return unless consent_refused? || health_answers.empty?
-    self.health_answers = []
-  end
-
+  # Because there are branching paths in the consent form journey, fields
+  # sometimes get set with values that then have to be deleted if the user
+  # changes their mind and goes down a different path.
   def reset_unused_fields
     self.common_name = nil unless use_common_name?
 
     self.contact_method = nil unless ask_for_contact_method?
     self.contact_method_other = nil unless contact_method_other?
 
-    self.reason = nil unless consent_refused?
-    self.reason_notes = nil unless consent_refused?
+    if consent_refused?
+      self.gp_response = nil
```

Comment:
I thought Ruby must have a `.clear` for this ... and looks like it does. Completely optional, though, it's just a teensy bit of sugar.

```suggestion
      self.gp_response.clear
```